### PR TITLE
TE-3842: Added sniffer-disable global command

### PR DESCRIPTION
**Pull request by supproduction**
https://spryker.atlassian.net/browse/TE-3842
https://spryker.atlassian.net/browse/TE-3717 - CR

### CODE REVIEW

File: rules/correct-typescript-component-structure.js [hunk 1]
```diff
@@ -1,26 +1,29 @@
...
 
             if (!files.typescript || !files.typescript.exists || !files.typescript.api.external) {
                 return;
             }
 
-            const { typescript } = files;
-            const { external } = typescript.api;
-            const { classes } = external;
-            const { name: typescriptFileName } = typescript;
+            const { name: typescriptFileName, disabledSnifferRules, api: { external: { classes } } } = files.typescript;
```
Comment (KostKravchenko) [inline rules/correct-typescript-component-structure.js]: please return step-by-step defragmentation

File: rules/correct-twig-component-structure.js [hunk 2]
```diff
@@ -21,14 +21,29 @@ module.exports = class extends Rule {
...
-            const { twig } = files;
-            const { name: twigFileName, content, api } = twig;
-            const configDefinition = api.external.definitions.filter(definition => definition.name === 'config')[0];
-            const blocks = api.external.blocks;
+            const {
+                disabledSnifferRules,
+                name: twigFileName,
+                content,
+                api: { external: { definitions, blocks } }
+            } = files.twig;
```
Comment (KostKravchenko) [inline rules/correct-twig-component-structure.js]: Please return step-by-step defragmentation

File: rules/correct-typescript-component-structure.js (hunk 1 above)
Comment (KostKravchenko) [inline rules/correct-typescript-component-structure.js]: Please return step-by-step defragmentation

**Review by KostKravchenko — CHANGES_REQUESTED** [review]

File: src/sniffer/test-outcome.ts [hunk 3]
```diff
@@ -27,6 +29,10 @@ export class TestOutcome {
         this.errorMessages.push(message);
     }
 
+    errorMessage(message: string, type: string, name: string, path: string): string {
```
Comment (tolerants) [inline src/sniffer/test-outcome.ts]: Suggest to rename into `formatMessage`.

File: rules/correct-scss-component-structure.js [hunk 4]
```diff
@@ -8,19 +7,23 @@ module.exports = class extends Rule {
...
             if (!files.sass || !files.sass.exists) {
                 return;
             }
 
-            const { sass } = component.files;
-            const { mixins } = sass.api.external;
-            const { name: scssFileName } = sass;
+            const { disabledSnifferRules, name: scssFileName, api: { external: { mixins } }} = files.sass;
+
+            if (isSnifferDisabled(disabledSnifferRules, this.getName())) {
```
Comment (tolerants) [inline rules/correct-scss-component-structure.js]: Suggest to implement a decorator for this check.

File: rules/correct-twig-component-structure.js [hunk 5]
```diff
@@ -21,14 +21,29 @@ module.exports = class extends Rule {
...
+
+            if (isSnifferDisabled(disabledSnifferRules, this.getName())) {
+                return;
+            }
+
+            const configDefinition = definitions.filter(definition => definition.name === 'config')[0];
```
Comment (tolerants) [inline rules/correct-twig-component-structure.js]: Suggest to add a check if there is only one `config` object. And also the same about `data` object.

**Review by tolerants — CHANGES_REQUESTED** [review]

**Review by tolerants — APPROVED** [review]

**Review by KostKravchenko — APPROVED** [review]

**Pull request merged**